### ISSUE TITLE
Bad Montserrat font version (firefox bug if using font from google fonts)

### ISSUE THREAD

**Issue by FragsterAt** [quasarframework/quasar#15639]
### Description

![изображение](https://user-images.githubusercontent.com/20900647/228488060-3355598d-3393-40d2-883f-f616856356f3.png)

Montserrat font has very bad readability in firefox, when using it version from google fonts package or cdn. There is updated version without this bug on https://fontsource.org/ , that also have good packages with very simple including:
```sass
@import "@fontsource/montserrat/200.css"
@import "@fontsource/montserrat/400.css"
@import "@fontsource/montserrat/500.css"
@import "@fontsource/montserrat/600.css"
@import "@fontsource/montserrat/700.css"
```

### Flavour

None

### Areas

Style & Identity (quasar)

**Comment by rstoenescu** [comment]
Thank you for this ticket.
We haven't bumped into this issue, but this is a good opportunity to understand why some of the devs were reporting "bad typography".

Next docs deployment will have the fixed montserrat version.

**Comment by FragsterAt** [comment]
@rstoenescu 

fonts, attached from google fonts:
![изображение](https://user-images.githubusercontent.com/20900647/229442804-39c19a4f-abf7-45fe-acd8-615fb1b60ea4.png)


fonts, attached from fontsource:
![изображение](https://user-images.githubusercontent.com/20900647/229442628-d5474232-e655-4944-b2a8-31322391f376.png)

**Comment by rstoenescu** [comment]
Docs have been updated.
Can you confirm that it now works as expected? Just to be sure that you're looking at latest and greatest, please remove the PWA and clear the browser cache.

**Comment by FragsterAt** [comment]
> Docs have been updated.
> Can you confirm that it now works as expected? Just to be sure that you're looking at latest and greatest, please remove the PWA and clear the browser cache.

The problem seems to be more complex. Firefox has turned off support of variable fonts in OS < win 10 (win server < 2019), so if you user of this OSes (like me with win server 2016), you see thin montserrat, if site use variable version of this font. Fontsource use fixed-weight version, so it forks fine on this OSes.
This is also actual for macOS < 10.13
see notes at https://caniuse.com/variable-fonts

**Comment by rstoenescu** [comment]
Try number 2. Docs deploying.

**Comment by rstoenescu** [comment]
It's been deployed with fallback. Can you check again, pls?
Thanks for all your help!

**Comment by FragsterAt** [comment]
Now it is fine! Thanks a lot!
![изображение](https://user-images.githubusercontent.com/20900647/230390969-64be55d8-44e7-4c5e-b708-8642a5fd95d7.png)

**Comment by rstoenescu** [comment]
Thank YOU for the detailed report here. There were many devs that complained about "bad typography" and we couldn't understand what they meant because for all of us it looked fine. Since no screenshots were attached we just thought that they didn't like the font itself in the first place. But due to your ticket, we could finally nail the problem and fix it.

**Comment by FragsterAt** [comment]
It returns!

<img width="1178" height="170" alt="Image" src="https://github.com/user-attachments/assets/77ed663b-da65-46e6-ba81-c04bc7f23ba7" />

**Comment by rstoenescu** [comment]
Thanks for reporting it again. I dug into it and this time the cause is in Firefox, not in the docs.

**What changed**

Firefox 155 (released September 1st) shipped [Bug 1872781](https://bugzilla.mozilla.org/show_bug.cgi?id=1872781), which removed the `layout.css.font-variations.enabled` pref. Up to Firefox 154 that pref was locked to `false` on platforms without variable font support (Windows 10 before build 1709 and Windows Server 2016), and with the pref off Firefox treated `format("woff2-variations")` as an unknown format and skipped that source. That is what made the 2023 fix work: on your machine Firefox fell through to the static per-weight Montserrat files.

In Firefox 155 the `woff2-variations` hint always maps to plain woff2 and `tech(variations)` is no longer rejected either, so the variable file gets loaded. The platform check that decides whether the weight axis can be applied is still there though, so on Windows Server 2016 the font is drawn at its default instance, which for Montserrat is weight 100 (Thin). That is the screenshot you posted.

**What I verified**

- The font files served by quasar.dev are byte-identical to the ones deployed for the 2023 fix, and the `@font-face` rules still list the static file as a fallback.
- Firefox 153 with the variation axis disabled still picks the static files and renders correctly with the current CSS.
- Forcing the variable file to load with the axis disabled reproduces your screenshot exactly.
- `@supports font-tech(variations)` also returns `true` on those platforms in Firefox 155, so there is no CSS-side way left to detect this. It affects every site that relied on the `-variations` format hint for its fallback, not just quasar.dev.

**Decision**

The only fixes on our side are serving static Montserrat files to everyone, or shipping a re-instanced variable file with a Regular default (which still renders every weight as Regular on your platform). Both cost every visitor extra bytes for a platform that Microsoft retires in early 2027, so we are not going to change the docs for this. The proper fix belongs in Firefox.

For your own apps, using fixed-weight files as you suggested in your first message (e.g. fontsource) keeps working regardless of the browser's variable font support.

**Comment by rstoenescu** [comment]
@FragsterAt since you have a machine that reproduces this and we don't, would you be up for filing it with Mozilla? A report from someone who can attach a real screenshot carries more weight than our emulation. Here is everything you need, feel free to paste it as is:

- File at https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=Layout%3A%20Text%20and%20Fonts
- Set **Regressed by** to `1872781` and add the `regression` keyword. That will notify the patch author automatically.
- Mention your exact Firefox version and Windows build (Server 2016 / Windows 10 before 1709).

Suggested summary:

> Firefox 155 loads `format("woff2-variations")` / `tech(variations)` sources on platforms without variable font support and renders them at the default instance

Suggested description:

> Since Firefox 155 (Bug 1872781 removed the `layout.css.font-variations.enabled` pref), an `@font-face` with `src: url(var.woff2) format("woff2-variations"), url(static.woff2) format("woff2")` loads the variable file on Windows Server 2016 / Windows 10 before 1709, where `gfxWindowsPlatform::CheckVariationFontSupport()` returns false. Up to Firefox 154 the pref was locked to false on those platforms and the `-variations` source was skipped, so the static fallback was used. Now the variable font is loaded but the weight axis is never applied (`gfxFontEntry::GetVariationsForStyle` still bails on `HasVariationFontSupport()`), so all text renders at the font's default instance. For Montserrat that is weight 100 (Thin), see screenshot. `@supports font-tech(variations)` also evaluates to true on these platforms, so there is no way for a site to detect this in CSS. CSS Fonts 4 requires skipping a source whose `tech()` the UA does not support. Example page: https://quasar.dev (header text). Expected: either skip the `-variations` / `tech(variations)` source when `HasVariationFontSupport()` is false, or apply the axis.

If you post the link here I'll follow it. Thanks again for sticking with this one.
